interactive_engine/groot/src/main/java/com/alibaba/graphscope/groot/coordinator/BackupManager.java

```diff
@@ -0,0 +1,542 @@
+            this.backupCreationScheduler.shutdown();
+            try {
+                this.backupCreationScheduler.awaitTermination(3000L, TimeUnit.MILLISECONDS);
+            } catch (InterruptedException e) {
+                // Ignore
+            }
+            this.backupCreationScheduler = null;
+        }
+        if (this.backupGcScheduler != null) {
+            this.backupGcScheduler.shutdown();
+            try {
+                this.backupGcScheduler.awaitTermination(3000L, TimeUnit.MILLISECONDS);
+            } catch (InterruptedException e) {
+                // Ignore
+            }
+            this.backupGcScheduler = null;
+        }
+        if (this.autoCommitScheduler != null) {
+            this.autoCommitScheduler.shutdown();
+            try {
+                this.autoCommitScheduler.awaitTermination(3000L, TimeUnit.MILLISECONDS);
+            } catch (InterruptedException e) {
+                // Ignore
+            }
+            this.autoCommitScheduler = null;
+        }
+    }
+
+    public int createNewBackup() throws BackupException, IOException {
+        checkEnable();
+        int newGlobalBackupId = increaseGlobalBackupId();
+        boolean suc = this.backupCreationBuffer.offer(newGlobalBackupId);
+        if (!suc) {
+            throw new BackupException("add backup creation task to buffer failed, new global backupId: " + newGlobalBackupId);
+        }
+        this.backupCreationScheduler.schedule(this::processBackupCreationTasks, 0L, TimeUnit.SECONDS);
+        logger.info("submit new backup creation task with backupId #[" + newGlobalBackupId + "]");
+        return newGlobalBackupId;
+    }
+
+    public void deleteBackup(int globalBackupId) throws BackupException, IOException {
+        checkEnable();
+        if (!this.globalBackupIdToInfo.containsKey(globalBackupId)) {
+            logger.warn("try to delete unavailable backup #[" + globalBackupId + "], ignore");
+            return;
+        }
+        removeBackupInfo(globalBackupId);
+        logger.info("backup #[" + globalBackupId + "] deleted");
+    }
+
+    public void purgeOldBackups(int keepAliveNum) throws BackupException, IOException {
+        checkEnable();
+        if (keepAliveNum <= 0) {
+            throw new IllegalArgumentException("the input keepAliveNum should > 0, got " + keepAliveNum);
+        } else if (keepAliveNum >= this.globalBackupIdToInfo.size()) {
+            return;
+        }
+        int numPurged = purgeOldBackupInfo(keepAliveNum);
+        logger.info(numPurged + " old backups purged");
+    }
+
+    public void restoreFromBackup(int globalBackupId, String metaRestorePath, String storeRestorePath)
+            throws BackupException, IOException {
+        checkEnable();
+        if (!this.globalBackupIdToInfo.containsKey(globalBackupId)) {
+            throw new BackupException("backup [" + globalBackupId + "] to restore is unavailable");
+        }
+        restoreGraphMeta(globalBackupId, metaRestorePath);
```

Comment:
I think this will not work as expected, since data in kafka queue would be send to the store after restore.

Reply:
for gnn now